src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -435,22 +435,25 @@ async def build_pex(
 
     description = request.description
     if description is None:
-        if request.requirements.req_strings:
-            description = (
-                f"Building {request.output_filename} with "
-                f"{pluralize(len(request.requirements.req_strings), 'requirement')}: "
-                f"{', '.join(request.requirements.req_strings)}"
-            )
-        elif request.requirements.file_path:
-            description = (
-                f"Building {request.output_filename} from {request.requirements.file_path}"
-            )
-        elif request.requirements.file_content:
-            description = (
-                f"Building {request.output_filename} from {request.requirements.file_content.path}"
-            )
-        else:
+        if not request.requirements:
             description = f"Building {request.output_filename}"
+        else:
+            if request.requirements.file_path:
+                desc_suffix = f"from {request.requirements.file_path}"
+            elif request.requirements.file_content:
+                desc_suffix = f"from {request.requirements.file_content.path}"
+            else:
+                desc_suffix = (
+                    f"with {pluralize(len(request.requirements.req_strings), 'requirement')}: "
+                    f"{', '.join(request.requirements.req_strings)}"
+                )
+            if request.repository_pex:
+                description = (
+                    f"Extracting from {request.repository_pex.name} to build "
```

Comment:
@stuhood and I also proposed "subsetting", "trimming", and "scoping".

Reply:
I'd de-emphasize the resolve name a little bit in favor of the per-consumer name: maybe by flipping the order a bit:
```
Extracting 1 requirement for requirements.pex from lockfile.pex: pytest<6.3,>=6.0.1
```

Reply:
I originally had that ordering but inverted it so that it was easier to craft the message w/ all these conditionals, hah. But I agree that we should prioritize a clear message over simple code.